src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/reroute/actions/UpdateFlowStatusAction.java

```diff
@@ -66,18 +61,30 @@ protected void perform(State from, State to, Event event, FlowRerouteContext con
         stateMachine.saveActionToHistory(format("The flow status was set to %s", resultStatus));
     }
 
-    private String getFlowStatusInfo(FlowStatus flowStatus, FlowRerouteFsm stateMachine) {
+    private String getFlowStatusInfo(Flow flow, FlowStatus flowStatus, FlowRerouteFsm stateMachine) {
         String flowStatusInfo = null;
         if (!FlowStatus.UP.equals(flowStatus) && !flowStatus.equals(stateMachine.getOriginalFlowStatus())) {
             flowStatusInfo = stateMachine.getErrorReason();
             if (FlowStatus.DEGRADED.equals(flowStatus)) {
-                if (stateMachine.isIgnoreBandwidth()) {
-                    flowStatusInfo = IGNORED_BW_RE_ROUTE_FLOW_STATUS_INFO;
-                } else {
-                    flowStatusInfo = DEGRADED_FLOW_STATUS_INFO;
-                }
+                flowStatusInfo = getDegradedFlowStatusInfo(flow, stateMachine);
             }
         }
         return flowStatusInfo;
     }
+
+    private String getDegradedFlowStatusInfo(Flow flow, FlowRerouteFsm stateMachine) {
+        boolean ignoreBandwidth = stateMachine.isIgnoreBandwidth();
+        boolean ignoreLatency = stateMachine.getNewPrimaryPathComputationStrategy() != flow.getPathComputationStrategy()
+                || (flow.isAllocateProtectedPath()
+                && stateMachine.getNewProtectedPathComputationStrategy() != flow.getPathComputationStrategy());
+        if (ignoreBandwidth && ignoreLatency) {
```

Comment:
Why we are ding all this "computations" here? Can the code that makes a decision to assign "DEGRATED" status to the flow produce statusInfo field too? And use the same transport for this field i.e. `stateMachine.getErrorReason()`?

Reply:
"DEGRADED" flow status is assigned a few lines before this computations.